client/electron/custom_install_steps.nsh

```diff
@@ -1,170 +1,170 @@
-  ;    bogus values so we don't have to fetch the real time.
-  ;  - Because nsExec::ExecToStack yields "NSIS strings" strings suitable for inclusion in, for
-  ;    example, a MessageBox, e.g. "device not found$\ncommand failed", we must convert it to a
-  ;    string that Sentry will like *and* can fit on one line, e.g.
-  ;    "device not found\ncommand failed"; fortunately, StrFunc.nsh's StrNSISToIO does precisely
-  ;    this.
-  ;  - RELEASE and SENTRY_URL are defined in env.nsh which is generated at build time by
-  ;    {package,release}_action.sh.
-
-  ; TODO: Remove this once we figure out why/if breadcrumbs are being truncated.
-  Var /GLOBAL FAILURE_MESSAGE_LENGTH
-  StrLen $FAILURE_MESSAGE_LENGTH $1
-
-  ; http://nsis.sourceforge.net/Docs/StrFunc/StrFunc.txt
-  Var /GLOBAL FAILURE_MESSAGE
-  ${StrNSISToIO} $FAILURE_MESSAGE $1
-  ${StrRep} $FAILURE_MESSAGE $FAILURE_MESSAGE '"' '\"'
-
-  ${WinVerGetMajor} $R0
-  ${WinVerGetMinor} $R1
-  ${WinVerGetBuild} $R2
-
-  ; http://nsis.sourceforge.net/Inetc_plug-in#post
-  inetc::post '{\
-    "message":"could not install TAP device ($0)",\
-    "release":"${RELEASE}",\
-    "tags":[\
-      ["os", "Windows $R0.$R1.$R2"],\
-      ["error_message_length", "$FAILURE_MESSAGE_LENGTH"]\
-    ],\
-    "breadcrumbs":[\
-      {"timestamp":1, "message":"$FAILURE_MESSAGE"}\
-    ]\
-  }' /TOSTACK ${SENTRY_URL} /END
-
-  Quit
-
-  installservice:
-
-  nsExec::Exec install_windows_service.bat
-
-  nsExec::Exec "$SYSDIR\sc query OutlineService"
-  Pop $0
-  StrCmp $0 0 success
-  ; TODO: Trigger a Sentry report for service installation failure, too, and revisit
-  ;       the restart stuff in the TypeScript code.
-  MessageBox MB_OK "Sorry, we could not configure your system to connect to Outline. Please try \
-    running the installer again. If you still cannot install Outline, please get in touch with us \
-    and let us know that OutlineService failed to install."
-  Quit
-
-  success:
-
-!macroend
-
-; TODO: Remove the TAP device on uninstall. This is impossible to implement safely
-;       with the bundled tapinstall.exe because it can only remove *all* devices
-;       having hwid tap0901 and these may include non-Outline devices.
-!macro customUnInstall
-  nsExec::Exec "$SYSDIR\net stop OutlineService"
-  nsExec::Exec "$SYSDIR\sc delete OutlineService"
-!macroend
+; Copyright 2018 The Outline Authors
+;
+; Licensed under the Apache License, Version 2.0 (the "License");
+; you may not use this file except in compliance with the License.
+; You may obtain a copy of the License at
+;
+;      http://www.apache.org/licenses/LICENSE-2.0
+;
+; Unless required by applicable law or agreed to in writing, software
+; distributed under the License is distributed on an "AS IS" BASIS,
+; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+; See the License for the specific language governing permissions and
+; limitations under the License.
+
+!include StrFunc.nsh
```

Comment:
GitHub shows many (unchanged) changes in this file. Maybe you changed the line endings? Please double confirm, it should be `CRLF`.

Reply:
Let me build it.